src/status.rs

```diff
@@ -0,0 +1,95 @@
+//! A Status Report announces the station's current mission or any other single
+//! line status to everyone. The report starts with the '>' APRS Data Type Identifier.
+//! The report may optionally contain a timestamp.
+//!
+//! Examples:
+//! - ">12.6V 0.2A 22degC"              (report without timestamp)
+//! - ">120503hFatal error"             (report with timestamp in HMS format)
+//! - ">281205zSystem will shutdown"    (report with timestamp in DHM format)
+
+use std::convert::TryFrom;
+use std::io::Write;
+
+use AprsError;
+use EncodeError;
+use Timestamp;
+
+#[derive(PartialEq, Debug, Clone)]
+pub struct AprsStatus {
```

Comment:
The spec puts a length limit on the comment (with a different value depending on whether or not there is a timestamp), which we aren't currently checking anywhere. I'd probably go for making these fields private and adding getters for them and a `new` method that validates the length.

Reply:
For the encoding we could check if the timestamp format is correct and the total length is up to 62. But since the message also has a length limit (67 characters) we are not checking I think this would be a new issue for a new pull request.

Reply:
Yeah, that makes sense.